client/lib/domains/can-domain-owner-add-email.ts

```diff
@@ -0,0 +1,19 @@
+import { getCurrentUserCannotAddEmailReason } from 'calypso/lib/domains';
+import { EMAIL_WARNING_CODE_OTHER_USER_OWNS_EMAIL } from 'calypso/lib/emails/email-provider-constants';
+import type { ResponseDomain } from 'calypso/lib/domains/types';
+
+/**
+ * Determines if domain owner can add email (i.e. domain owner is also the email owner).
```

Comment:
What do you think of clarifying even more what that means?

```suggestion
 * Determines if the domain owner can add email (i.e. if the owner of the domain subscription is also the owner of the email subscription).
```